lib/mlir/Dialect/Clift/IR/CliftAttributes.cpp

```diff
@@ -307,59 +308,62 @@ mlir::Attribute mlir::clift::StructType::parse(AsmParser &parser) {
   return parseImpl<StructType>(parser, "union");
 }
 
+static bool isCompleteType(const mlir::Type Type) {
+  if (auto T = mlir::dyn_cast<mlir::clift::DefinedType>(Type)) {
+    auto Definition = T.getElementType();
+    if (auto D = mlir::dyn_cast<mlir::clift::StructType>(Definition))
+      return D.isDefinition();
+    if (auto D = mlir::dyn_cast<mlir::clift::UnionType>(Definition))
+      return D.isDefinition();
+    return true;
+  }
+
+  return true;
```

Comment:
Have a look at [this commit](https://github.com/revng/revng-c/pull/20/commits/565e42c5268695de4ba2db4b50dade42bbdbf56c) in my next PR.

Reply:
I checked, there are a few things that make sense, and some that are wrong. I left comments on the commit.

In any case pick that commit and make it part of this PR please, so that we merge things that are correct at the first shot.

And then we can fix the few issues that remain, that I've stressed in my comments there.

Reply:
Ok I'm closing this. Just pick the commit you linked above and bring it in this PR.